src/main/java/uk/gov/companieshouse/companyprofile/delta/mapper/CompanyProfileMapper.java

```diff
@@ -342,6 +344,62 @@ public void mapEnumsJurisdiction(@MappingTarget CompanyProfile target, CompanyDe
         target.setData(data);
     }
 
+    /**Maps Foreign Company Account type to string. */
+    @AfterMapping
+    public void mapForeignAccountType(@MappingTarget CompanyProfile target, CompanyDelta source) {
+        Data data = target.getData();
+        if (data.getForeignCompanyDetails() != null) {
+            ForeignCompanyDetails foreignCompanyDetails = data.getForeignCompanyDetails();
+            if (foreignCompanyDetails.getAccountingRequirement() != null) {
+                AccountingRequirement accountingRequirement = foreignCompanyDetails.getAccountingRequirement();
+
+                String foreignAccountType = source.getForeignCompany().getAccReqType();
+
+                HashMap<String,String> foreignAccountTypeMap = MapperUtils.getForeignAccountTypeMap();
+                accountingRequirement.setForeignAccountType(foreignAccountTypeMap
+                        .getOrDefault(foreignAccountType, null));
+                foreignCompanyDetails.setAccountingRequirement(accountingRequirement);
+                data.setForeignCompanyDetails(foreignCompanyDetails);
+                target.setData(data);
+            }
+        }
+    }
+
+    /**Maps Foreign Company terms of account publication. */
+    @AfterMapping
+    public void mapTermsOfAccountPublication(@MappingTarget CompanyProfile target, CompanyDelta source) {
+        Data data = target.getData();
+        if (data.getForeignCompanyDetails() != null) {
+            ForeignCompanyDetails foreignCompanyDetails = data.getForeignCompanyDetails();
+            if (foreignCompanyDetails.getAccountingRequirement() != null) {
+                AccountingRequirement accountingRequirement = foreignCompanyDetails.getAccountingRequirement();
+                String termsOfAccountPublication = source.getForeignCompany().getTermsOfAccountPublication();
+
+                HashMap<String,String> termsOfAccountPublicationMap = MapperUtils.getTermsOfAccountPublicationMap();
+                accountingRequirement.setTermsOfAccountPublication(termsOfAccountPublicationMap
+                        .getOrDefault(termsOfAccountPublication, null));
+                foreignCompanyDetails.setAccountingRequirement(accountingRequirement);
+                data.setForeignCompanyDetails(foreignCompanyDetails);
+                target.setData(data);
+            }
+        }
+    }
+
+    /**Maps Foreign Company Credit or Financial boolean. */
+    @AfterMapping
+    public void mapCreditOrFinancial(@MappingTarget CompanyProfile target, CompanyDelta source) {
+        Data data = target.getData();
+        if (data.getForeignCompanyDetails() != null) {
+            ForeignCompanyDetails foreignCompanyDetails = data.getForeignCompanyDetails();
+
+            String isCreditOrFinancial = source.getForeignCompany().getCreditOrFinancial();
+            foreignCompanyDetails.setIsACreditFinancialInstitution(!isCreditOrFinancial.equals("0"));
```

Comment:
This will map anything that isn't "0" to true. I think we want to map null to null and any string that isn't "1" to false